File: src/1436-Destination-City/1436.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_set<string> s;
        for (auto& path : paths) s.insert(path[0]);
        for (auto& path : paths) {
            if (!s.count(path[1])) return path[1];
        }
        return "";
    }
};

File: src/1436-Destination-City/1436.java
class Solution {
    public String destCity(List<List<String>> paths) {
        Set<String> set = new HashSet();
        for (List<String> path : paths) {
            set.add(path.get(0));
        }
        for (List<String> path : paths) {
            if (!set.contains(path.get(1))) return path.get(1);
        }
        return "";
    }
}